src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -2620,6 +2635,82 @@ namespace SimAirServingZones {
 
         } // End of Air Loop iteration
 
+        if (numAirLoopDOAS > 0) {
+            int index;
+            Real64 OAMassFLowrate = 0.0;
+            for (std::size_t loop = 0; loop < airloopDOAS.size(); ++loop) {
+                AirLoopHVACDOAS::AirLoopDOAS *thisAirLoopDOASObjec = &airloopDOAS[loop];
+                if (thisAirLoopDOASObjec->m_AirLoopDOASNum > -1) {
+                    index = thisAirLoopDOASObjec->m_AirLoopDOASNum;
+                } else {
+                    index = -1;
+                }
+                thisAirLoopDOASObjec->SimAirLoopHVACDOAS(FirstHVACIteration, index);
+                OAMassFLowrate += thisAirLoopDOASObjec->SumMassFlowRate;
+            }
```

Comment:
It looks like this whole loop could use range based looping:

``` c++
for (auto & thisAirLoopDOASObjec : airloopDOAS) { // <- let compiler create the reference and do the looping
  if (thisAirLoopDOASObjec.m_AirLoopDOASNum > -1) {
    index = thisAirLoopDOASObjec.m_AirLoopDOASNum;
  } else {
    index = -1;
  }
  thisAirLoopDOASObjec.SimAirLoopHVACDOAS(FirstHVACIteration, index);
  OAMassFLowrate += thisAirLoopDOASObjec.SumMassFlowRate;
}
```

There's no need to introduce a pointer reference here.  Even if you don't change the entire loop to a range based loop, it would be good to use a regular reference:

``` c++
for (std::size_t loop = 0; loop < airloopDOAS.size(); ++loop) {
  auto &thisAirLoopDOASObjec = airloopDOAS[loop]; // <- regular reference variable, not a pointer
  if (thisAirLoopDOASObjec.m_AirLoopDOASNum > -1) {
    index = thisAirLoopDOASObjec.m_AirLoopDOASNum;
  } else {
    index = -1;
  }
  thisAirLoopDOASObjec.SimAirLoopHVACDOAS(FirstHVACIteration, index);
  OAMassFLowrate += thisAirLoopDOASObjec.SumMassFlowRate;
}
```

I don't want to introduce this pattern into the code, so this will be the first thing in my review that I would actually like to see get addressed.  There may be other things as I go through, but this is the first thing.

Reply:
@Myoldmopar I am going to use a regular reference, so that the code change is less.